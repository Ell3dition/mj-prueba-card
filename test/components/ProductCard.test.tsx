import React from "react"
import renderer from "react-test-renderer"
import { ProductCard } from '../../src/components'
import { productUno } from "../data/product"


describe('ProductCard', ()=>{

    test('Debe de mostrar el componente correctamente', ()=>{

        const wrapper = renderer.create(
            <ProductCard product={productUno}  className="custom-class" >
        {        ()=>(
            <h1>Hello</h1>
        )
                }

            </ProductCard>
        )
        
        expect(wrapper.toJSON()).toMatchSnapshot()
    })


    test('debe de incrementar el contador ', () => {
        const wrapper = renderer.create(
            <ProductCard product={productUno}  className="custom-class" >
        {        ({count, increaseBy})=>(

                    <>
                        <h1>ProductCard</h1>
                        <span>{count}</span>
                        <button onClick={()=>increaseBy(1)}></button>
                    </>
        )
                }

            </ProductCard>
        )
        
       let tree = wrapper.toJSON()
       expect(tree).toMatchSnapshot();
    })
    

 
})