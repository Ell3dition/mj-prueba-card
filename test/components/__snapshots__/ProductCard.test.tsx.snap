// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe de mostrar el componente correctamente 1`] = `
<div
  className="productCard custom-class"
>
  <h1>
    Hello
  </h1>
</div>
`;

exports[`ProductCard debe de incrementar el contador  1`] = `
<div
  className="productCard custom-class"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
